### DISCUSSION TITLE
skill-ovos-youtube: uri times out

### DISCUSSION THREAD

**Discussion by wita-sec** [OpenVoiceOS/ovos-buildroot#277] [Q&A] [answered]
RPi4B configured for de-de
skill-ovos-youtube
uterance: spiele rob zombie (play rob zombie)
times out, however the uri's which the skill finds are all valid
playing these uris from the RPi4B browser works fine.

Do I need to configure something in OVOS, e.g. increase the timeout?

Here is an extract form audio.log

[20240921-youtube-playback-timeout.txt](https://github.com/user-attachments/files/17084676/20240921-youtube-playback-timeout.txt)

**Comment by JarbasAl** [comment]
can "mpv" play the urls? maybe you are missing some dependency like youtube-dl?

**Reply by wita-sec** [reply]
have installed youtube-dl
after that mpv works fine: 
mpv --no-video https://www.youtube.com/watch?v=nfWlot6h_JM

**Reply by JarbasAl** [reply]
Then it should also work via ovos, as it just delegates the playback to mpv

**Comment by JarbasAl** [comment]
alternatively, is https://github.com/OpenVoiceOS/ovos-ocp-youtube-plugin installed?

**Reply by wita-sec** [reply]
mpv works fine:
mpv --no-video https://www.youtube.com/watch?v=nfWlot6h_JM

I installed ovos-ocp-youtube-plugin 
after that, "play rob zombie" could not find these tracks any longer.
I uninstalled ovos-ocp-youtube-plugin, however this problem perstists. 
I made sure to restart ovos after each of the install steps. 

I will investigate further next week, no time left for this weekend.

Thank you for your advice so far.

**Reply by JarbasAl** [reply]
The OCP plugin would extract the real stream allowing it to play without expecting mpv to extract the real url, but it should not affect the searching of results at all... 

Wonder if YouTube is doing changes or testing things only in some locations (YouTube is notorious for not rolling out changes everywhere at once)

**Reply by wita-sec** [reply]
Have shutdown ovos
Have again installed ovos-ocp-youtube-plugin
Have started ovos

Timeout problem still persists.
However the skill skill-ovos-youtube was able to find the tracks on youtube.
Maybe in my earlier test I was too inpatient when investigating audio.log.

Here comes an extract of  the newest audio.log 
[20240922-youtube-playback-timeout.txt](https://github.com/user-attachments/files/17091052/20240922-youtube-playback-timeout.txt)

**Reply by JarbasAl** [reply, marked as answer]
You mention installing 3 weeks ago, can I ask you to reinstall or update? 3 weeks ago we were in alpha, currently we have stable packages that include fixes related to media playback. Perhaps updating OCP and ovos audio is enough if you want to avoid doing a full reinstall. But please make sure you are on latest stable packages

**Reply by wita-sec** [reply]
I have shutdown ovos, then 
rerun ovos-installer,
It detected by current installation. 
I walked through the dialogs and kicked off installation.
Then restarted ovos.

I had some trouble, because pulseaudio was no longer working, and ovos failed to open pulseaudio.socket.
I needed to reinstall and configure pulseaudio again. 

Now the youtube skill works fine.

These are the most recent audio.log records.
[20240922-youtube-playback-timeout.txt](https://github.com/user-attachments/files/17100406/20240922-youtube-playback-timeout.txt)

Thank you for help.

If you like I can open a new ticket and send you the ovos-installer logs.
Just give me a hint where i can find theses logs.

**Reply by JarbasAl** [reply]
open an issue in ovos-installer repo for that please
